app/validation/validator.py

```diff
@@ -154,13 +154,15 @@ def validate_routing_on_answer_options(self, block, answer, answer_ids):
 
             for rule in block['routing_rules']:
                 if 'goto' in rule and 'when' in rule['goto'].keys():
-                    when = rule['goto']['when']
-                    if 'id' in when[0]:
-                        if not self.validate_when_rule(when, answer_ids):
-                            answer_errors.append(self._error_message('The answer id - {} in the "when" clause for {} '
-                                                                     'does not exist'.format(when[0]['id'], block['id'])))
-                        if when[0]['id'] == answer['id'] and when[0]['value'] in options:
-                            options.remove(when[0]['value'])
+                    when_clause = rule['goto']['when']
+                    valid_when_clause = self.validate_when_rule(when_clause, answer_ids, block['id'])
```

Comment:
Nitpick - since this method returns a tuple I'd suggest unpacking it for clarity rather than having to index it.

I.e.
```
is_valid, error_message = self.validate_when_rule(when_clause, answer_ids, block['id'])
if not is_valid:
   answer_errors.append(self._error_message(error_message))
```

Reply:
very good point 👍